bot/converters.py

```diff
@@ -267,37 +268,60 @@ async def convert(self, ctx: Context, arg: str) -> int:
         return snowflake
 
 
-class SourceConverter(Converter):
-    """Convert an argument into a help command, tag, command, or cog."""
+class SourceTransformer(app_commands.Transformer):
+    """Transform an argument into a tag, command or cog."""
 
-    @staticmethod
-    async def convert(ctx: Context, argument: str) -> SourceType:
-        """Convert argument into source object."""
-        if argument.lower() == "help":
-            return ctx.bot.help_command
+    async def transform(self, interaction: discord.Interaction, cog_command_or_tag: str) -> SourceType:
+        """Transform cog_command_or_tag into source object."""
+        if cog_command_or_tag.lower() == "help":
+            return interaction.client.help_command
 
-        cog = ctx.bot.get_cog(argument)
+        cog = interaction.client.get_cog(cog_command_or_tag)
         if cog:
             return cog
 
-        cmd = ctx.bot.get_command(argument)
+        cmd = interaction.client.get_command(cog_command_or_tag)
         if cmd:
             return cmd
 
-        tags_cog = ctx.bot.get_cog("Tags")
+        slash_cmd = interaction.client.tree.get_command(cog_command_or_tag)
+        slash_cmd = slash_cmd if not isinstance(slash_cmd, app_commands.ContextMenu) else None
+        if slash_cmd:
+            return slash_cmd
+
+        tags_cog = interaction.client.get_cog("Tags")
         show_tag = True
```

Comment:
it looks like show_tag isn't used anywhere, can we remove it then ?

Reply:
It is see https://github.com/python-discord/bot/pull/2446/files#diff-a39b0dd807dd61d8677c3fd3c20b409abf4c036c6dbdab15288bd592c3b843ebR304

Reply:
Oh sorry for that, it got mixed up when i changed to code to try exception related stuff